src/core/Tree/TreeNode.tsx

```diff
@@ -64,17 +48,13 @@ export type TreeNodeProps = {
  */
 export const TreeNode = (props: TreeNodeProps) => {
   const {
-    title,
-    caption,
-    isExpanded = false,
-    isActive,
+    label,
+    sublabel,
     depthLevel = 0,
```

Comment:
Maybe `depthLevel` can also be moved to the Context? What do you think @mayank99 @veekeys ?

Reply:
So wrap TreeNode children with context and it will override parent context.

Reply:
Added `nodeDepth` to context

Reply:
I think it might be better to use a separate context for this

Reply:
I think depth in context might be alright.. not sure, maybe technically better have separate, but..if it works, maybe not a big deal..